js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx

```diff
@@ -0,0 +1,335 @@
+import {useQuery} from '@apollo/client';
+import {Box, Colors, Heading, Icon, Page, Spinner, TextInput} from '@dagster-io/ui-components';
+import {Link, useParams} from 'react-router-dom';
+import styled from 'styled-components';
+
+import {AssetGlobalLineageButton, AssetPageHeader} from './AssetPageHeader';
+import {ASSET_CATALOG_TABLE_QUERY} from './AssetsCatalogTable';
+import {AssetTableFragment} from './types/AssetTableFragment.types';
+import {
+  AssetCatalogTableQuery,
+  AssetCatalogTableQueryVariables,
+} from './types/AssetsCatalogTable.types';
+import {useTrackPageView} from '../app/analytics';
+import {TagIcon} from '../graph/OpTags';
+import {useDocumentTitle} from '../hooks/useDocumentTitle';
+import {useLaunchPadHooks} from '../launchpad/LaunchpadHooksContext';
+import {ReloadAllButton} from '../workspace/ReloadAllButton';
+import {repoAddressAsHumanString, repoAddressAsURLString} from '../workspace/repoAddressAsString';
+import {RepoAddress} from '../workspace/types';
+
+const ViewAllAssetsLink = () => (
+  <Link to="/assets">
+    <Box flex={{gap: 4}}>View all</Box>
+  </Link>
+);
+
+type AssetCountsResult = {
+  countsByOwner: Record<string, number>;
+  countsByComputeKind: Record<string, number>;
+  countPerAssetGroup: CountPerGroupName[];
+  countPerCodeLocation: CountPerCodeLocation[];
+};
+
+type GroupMetadata = {
+  groupName: string;
+  repositoryLocationName: string;
+  repositoryName: string;
+};
+
+type CountPerGroupName = {
+  assetCount: number;
+  groupMetadata: GroupMetadata;
+};
+
+type CountPerCodeLocation = {
+  repoAddress: RepoAddress;
+  assetCount: number;
+};
+
+function buildAssetCountBySection(assets: AssetTableFragment[]): AssetCountsResult {
+  const assetCountByOwner: Record<string, number> = {};
+  const assetCountByComputeKind: Record<string, number> = {};
+  const assetCountByGroup: Record<string, number> = {};
+  const assetCountByCodeLocation: Record<string, number> = {};
+
+  assets.forEach((asset) => {
+    if (asset.definition) {
+      asset.definition.owners.forEach((owner) => {
+        const ownerKey = owner.__typename === 'UserAssetOwner' ? owner.email : owner.team;
+        assetCountByOwner[ownerKey] = (assetCountByOwner[ownerKey] || 0) + 1;
+      });
+
+      const computeKind = asset.definition.computeKind;
+      if (computeKind) {
+        assetCountByComputeKind[computeKind] = (assetCountByComputeKind[computeKind] || 0) + 1;
+      }
+
+      const groupName = asset.definition.groupName;
+      const locationName = asset.definition.repository.location.name;
+      const repositoryName = asset.definition.repository.name;
+
+      if (groupName) {
+        const metadata: GroupMetadata = {
+          groupName,
+          repositoryLocationName: locationName,
+          repositoryName,
+        };
+        const groupIdentifier = JSON.stringify(metadata);
```

Comment:
check out `groupIdForNode`

Reply:
Looks like that type relies on the asset node being the return type of a certain graphQL fragment. One option would be refactoring that function to accept a more generic type?

The only purpose of this `JSON.stringify` is to be able to key a dictionary by asset group metadata. Should I instead try using Map or something?

Reply:
oh usually I update the function's type to be only the things the function actually uses.

eg: `Pick<OriginalType, 'list' | 'of_keys' | 'to_pick'>`

Reply:
Ahhh that's a good tip, thanks for explaining

We still need to be able to unpack the group identifier back into the object though, would it make sense to keep using JSON to avoid having to write a reverse of `groupIdForNode`?

Reply:
I think JSON is fine since its only used locally